recipes-core/date-sync/init

```diff
@@ -11,66 +11,96 @@
 PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
 NAME=date-sync
 DESC="Date Sync"
+DAEMON=/usr/sbin/ntpd
 LOGFILE=/var/log/date-sync.log
+LOGFILE_MONITOR=/var/log/date-sync_monitor.log
 PIDFILE=/var/run/date-sync.pid
-
-# Configuration
+PIDFILE_MONITOR=/var/run/date-sync_monitor.pid
+SYSTEM_API_FIFO=/var/volatile/system-api.fifo
 NTP_SERVER="pool.ntp.org"
-SYNC_INTERVAL=60  # Run every 60 seconds (1 minute)
-MAX_LOG_SIZE=524288  # 512KB (0.5MB) in bytes
 
 log() {
-    echo "$(date +"%Y-%m-%d %H:%M:%S"): $1" >> $LOGFILE
-    if [ -f "$LOGFILE" ] && [ $(stat -c %s "$LOGFILE") -gt $MAX_LOG_SIZE ]; then
-        tail -n 1000 "$LOGFILE" > "${LOGFILE}.tmp" && mv "${LOGFILE}.tmp" "$LOGFILE"
-    fi
-}
-
-sync_time() {
-    ntpd_output=$(ntpd -n -q -d -p $NTP_SERVER 2>&1)
-    if [ $? -eq 0 ]; then
-        offset=$(echo "$ntpd_output" | grep "offset:" | tail -n 1 | sed 's/.*offset:\([^ ]*\).*/\1/')
-        if [ -n "$offset" ]; then
-            log "Time synchronized with $NTP_SERVER successfully. Offset: $offset seconds"
-        else
-            log "Time synchronized with $NTP_SERVER successfully. Offset information not found in output."
-        fi
+    if [ -p $SYSTEM_API_FIFO ]; then
+        date_log() {
+            echo -n ""
+        }
     else
-        log "Time synchronization with $NTP_SERVER failed. Error output: $ntpd_output"
+        date_log() {
+            echo -n "$(date --iso-8601=seconds): "
+        }
     fi
+    echo "$(date_log)$1" | tee -a $SYSTEM_API_FIFO
 }
 
-run_daemon() {
+monitor_and_restart() {
     while true; do
-        sync_time
-        sleep $SYNC_INTERVAL
+        if ! pgrep -f "$DAEMON" > /dev/null; then
+            log "$NAME crashed. Restarting in 60 seconds..." >> ${LOGFILE_MONITOR}
+            sleep 60
+            start
+        fi
+        sleep 5
     done
 }
 
-start_daemon() {
-    log "Starting $DESC"
-    run_daemon &
-    echo $! > $PIDFILE
+start() {
+    log "Starting $NAME"
+    start-stop-daemon -S --make-pidfile --pidfile $PIDFILE \
+        --background --startas /bin/sh -- -c "exec \
+        ${DAEMON} -n -d -p $NTP_SERVER \
+        2>&1 | tee ${LOGFILE}"
```

Comment:
nit: should it be `tee -a ${LOGFILE} ` to make it append to the logfile instead?
question: wouldn't this run the daemon once to execute this command and exists afterwards? then the monitor will restart it again every 60 seconds. Is this the difference between this and previous 10 min interval? Because the Interval could be set at build time to 60 seconds instead of 600. We only chose 10 minutes to not hit any rate limit.
Did you make sure that running the ntpd -d -p - n actually continues ? to my knowledge it is provided from the busybox tools and it is limited a bit

Reply:
> should it be tee -a ${LOGFILE} to make it append to the logfile instead?
The only drawback of `tee` without `-a` is that the log will be overwritten on each service restart. But the logs are forwarded to the remote logs collector anyways.